hazelcast/src/main/java/com/hazelcast/query/impl/predicates/AbstractPredicate.java

```diff
@@ -175,7 +175,7 @@ public void readData(ObjectDataInput in) throws IOException {
 
     @Override
     public int getFactoryId() {
-        return PREDICATE_DS_FACTORY_ID;
+        return Factory.PREDICATE_DS.getDefaultFactoryId();
```

Comment:
I'm not sure about this change.

1: it is a lot less informative because it isn't easy to spot the actual value being returned.

2: it is more verbose since there is a lot more going on than with the original value.

3: I have no idea about the performance implications. 

This part of the PR I'm not agreeing with.

Reply:
> I have no idea about the performance implications

how would you recommend assessing the performance implications? Is there a specific serialisation simulation you had in mind?

Reply:
This code should be reverted so that you are referencing a static const.

Reply:
[I've addressed this](https://github.com/hazelcast/hazelcast/pull/25247#discussion_r1303166087).